marconi-chain-index/README.md

```diff
@@ -1,178 +1,99 @@
-   * Weigh considerations for in-memory and on-disk data. The query function produces a result by merging the in-memory and on-disk data. If you are concerned only with on-disk data, then you can query the database directly.
+```
+$ nix build .#marconi-chain-index
+```
 
-3. Customize a function that stores buffered data. While there is no connection to any storage mechanism, there is a nice API is available, and you are encouraged to change it to suit your needs. Changing it is very simple.
+Or you may run from anywhere:
 
-Marconi uses the `streaming` library so you can make use of its many combinators.
+```
+$ nix build github:input-output-hk/plutus-apps#marconi-chain-index
+```
 
-## Documentation
+Both commands will produce a `result` directory containing the executable
+`result/bin/marconi-chain-index`.
 
-When it exists, we can link to our readthedocs user documentation.
+### Cabal+Nix build
 
-* User Guide
-* Reference Guide
-* Tutorial
-* Example Code
+To build `marconi-chain-index` from the source files, use the following commands:
 
-## Intended Use Cases
+```sh
+git clone git@github.com:input-output-hk/plutus-apps.git
+nix develop
+cabal clean && cabal update # Optional, but makes sure you start clean
+cabal build marconi-chain-index
+```
 
-Description of its primary intended use cases.
+The above process will build the executable in your local environment at this location:
 
-* Sync with the Cardano blockchain (private/public testnet or mainnet) by reading all blocks from the genesis block to the current tip
-* Index the syncing information based on the user’s predefined indexing
-* Query the indexed information based on the user’s predefined queries
+```sh
+cabal exec -- which marconi-chain-index
+```
 
-### Use Case 1
+Or you can run the executable directly with:
 
-### Use Case 2
+```sh
+cabal run marconi-chain-index:exe:marconi-chain-index -- --help
+```
 
-### Use Case 3
+## Command line summary
 
-## Example Queries
+Run `marconi-chain-index`, `$(cabal exec -- which marconi-chain-index) --help` or `cabal run marconi-chain-index:exe:marconi-chain-index -- --help` for a general synopsis of the command line options depending on your installation method.
 
-Can we provide example queries?
+## How to run
 
-## Architecture
+We are assuming that:
 
-* Provide a compelling diagram
+* you have a local running cardano-node instance.
+* you've set the following environment variables:
+  * `CARDANO_NODE_SOCKET_PATH`: socket path of your local cardano-node instance
+  * `MARCONI_DB_DIRECTORY`: directoy in which to create the various SQLite database files
 
-## Architecture Decision Records
+The most minimal way to run the executable is as follows:
 
-* Records of decisions that were made by the team.
-* Discuss alternatives.
-* Why did we choose one design over another?
-   * Konstantinos has some for Plutus Apps.
-   * Radu has some for Marconi.
+```sh
+marconi-chain-index \
```

Comment:
I'm not a fan of the golden tests that we have for the cli, but now that we do have them let's link to them?

Reply:
We don't have golden tests for the `marconi-chain-index` CLI, and it's a bit different than the `marconi-sidechain` one.

Reply:
I made an argument here https://github.com/input-output-hk/plutus-apps/pull/960#discussion_r1109924411 about why we would want golden tests.

What don't you like exactly? What can we do different which can achieve the same benefits as the golden tests that we have.

Reply:
> We don't have golden tests for the `marconi-chain-index` CLI, and it's a bit different than the `marconi-sidechain` one.

We added CLI golden test to [marconi-shain-index ](https://github.com/input-output-hk/plutus-apps/blob/main/marconi-chain-index/test/Spec/Golden/Cli/marconi-chain-index___help.help)

Reply:
@kayvank Thanks, added a link to the golden file. However, in this case, I want to show an example off how to start the command in the README, and the golden file doesn't provide that. We can probably extend it, but for now, let's keep it as is.

Reply:
Regarding not liking golden tests for CLI: the CLI text that an executable prints are directly generated from the CLI definitions that we defined by the help of optparse-applicative library and tests for that process already exist in optparse-applicative. I.e if we don't want to change the CLI then we should just ...not change the CLI definitions. And when we *do* want to change the CLI definitions then we now have *two* files to change.